### PR TITLE
C#: Fix `global.json` and `packages.config` lookup

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DotNet.cs [hunk 1]
```diff
@@ -143,7 +143,7 @@ private static BuildScript DownloadDotNet(IBuildActions actions, ILogger logger,
             // See https://docs.microsoft.com/en-us/dotnet/core/tools/global-json
             var versions = new List<string>();
 
-            foreach (var path in files.Where(p => p.EndsWith("global.json", StringComparison.Ordinal)))
+            foreach (var path in files.Where(p => p.EndsWith(Path.DirectorySeparatorChar + "global.json", StringComparison.Ordinal)))
```
Comment (michaelnebel) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DotNet.cs:146]: Did you run into a case where a file was name `myglobal.json` or something similar?
nit: Maybe consider using `Path.GetFileName`? (same comment for the other similar change)

**Review by michaelnebel — APPROVED** [review]
LGTM!

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DotNet.cs (hunk 1 above)
Comment (tamasvajk) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DotNet.cs:146]: Thanks for suggesting `Path.GetFileName`, that handles two other cases too. I'll change it...

I was putting together an integration test, and temporarily renamed the `packages.config` to `x_packages.config`, and then I was surprised that the packages were still picked up by the dependency manager.

**Review by michaelnebel — APPROVED** [review]
LGTM!

**Pull request merged**